Template 01 list created (#12)
OntimizeWeb/ontimize-web-templates: templates/list/01-list/README.md

## How to use

<br/>

1. Download and put `list` folder in src/app/main/ and `i18n` into assets/

<br/>

<pre>
─src
    ├───app
    |   ├───main
    |   |   └───list
    │   ├───...
    │   │
    ├───assets
    │   ├───css
    │   └───i18n
    |       ├───en.json
    |       └───es.json
    │
    │
    └───..

</pre>

> 2. Check if all the fields that have `@` on it's prefix and sufix are concording to your application data

Reviewer: ```suggestion
2. Check if all the fields that have `@` on it's prefix and suffix are concording to your application data
```

    controls="no" fxFlex.gt-lg="list-accounts-60" pageable="yes" fxFill>
```

<br/>

By:
```html
 <o-list #listAccount service="yourservice" entity="yourentity" keys="yourkey"
    columns="yourcolumn1;yourcolumn2"
    controls="no" fxFlex.gt-lg="list-accounts-60" pageable="yes" fxFill>
```

3. Configure your filter component setting your inputs and custom ATTR to build the filter. Example of how filter builder works at the end of the readme.

4. Add the translations you want to use on your app ​​to the `en.json` and `es.json` files of your project

<br/>

5. Configure routing in `main-routing.module.ts`

<br/>

```js
import { NgModule } from '@angular/core';
import { ExtraOptions, RouterModule, Routes } from '@angular/router';

export const routes: Routes = [
  ...
  { path: 'list', loadChildren: () => import('./list/list.module').then(m => m.ListModule) }
];

const opt: ExtraOptions = {
  enableTracing: false
  // true if you want to print navigation routes
};

@NgModule({
  imports: [RouterModule.forRoot(routes, opt)],
  exports: [RouterModule],
  providers: []
})
export class MainRoutingModule { }
```

<br/>

> 6. Add the module declaration in the `main.module.ts` file

Reviewer: Point 6 is not necessary

<br/>

> 7. To configure the theme on the list component it is necessary to include `list-home.theme.scss` in `app.scss`

Reviewer: ```suggestion
7. To configure the theme on the list component it is necessary to include `list.theme.scss` in `app.scss`
```

<br/>

```scss
@use 'theme.scss' as theme;
@use 'ontimize-web-ngx/theming/ontimize-style.scss';
...
@include ontimize-style.ontimize-theme-styles(theme.$theme);

> + @import '../../app/list/list-home.theme.scss';

Reviewer: ```suggestion
+ @import '../../app/list/list.theme.scss';
```

@mixin app-themes($theme) {
...
+ @include list-theme($theme);
...
}

.o-dark {
...
  + @include app-themes(theme.$dark-theme);
...
}

/*
* Propagate theme to screen styles definition.
*/
@include app-themes(theme.$theme);

/*
* Other app styles
*/

```

## LEARN MORE

<br/>

* **Dark and light mode** https://ontimizeweb.github.io/docs/v15/customize/theming/#dark-and-light-primary-variants

* **OTranslateService** https://ontimizeweb.github.io/docs/v15/guide/otranslateservice/overview

* **Filter builder** https://ontimizeweb.github.io/docs/v15/components/data/filterbuilder/overview

> * **List component** https://ontimizeweb.github.io/docs/v15/components/data/list/overview

Reviewer: ```suggestion
* **List component** https://ontimizeweb.github.io/docs/v15/components/data/list/overview
* ** MediaMatcher** https://material.angular.io/cdk/layout/overview
```

## How to use

<br/>

1. Download and put `list` folder in src/app/main/ and `i18n` into assets/

<br/>

<pre>
─src
    ├───app
    |   ├───main
    |   |   └───list
    │   ├───...
    │   │
    ├───assets
    │   ├───css
    │   └───i18n
    |       ├───en.json
    |       └───es.json
    │
    │
    └───..

</pre>

> 2. Configure the `o-grid`, you must to modify the values ​​between `@` for the inputs `attr`, `service`, `entity`, `columns`, `quick-filter-columns`, `sort-columns`, `keys` . For more information consult the following url https://ontimizeweb.github.io/docs/v15/components/data/grid/overview

Reviewer: ```suggestion
2. Configure the `o-list`, you must to modify the values ​​between `@` for the inputs `attr`, `service`, `entity`, `columns` and `keys` . For more information consult the following url https://ontimizeweb.github.io/docs/v15/components/data/list/overview
```

<br/>

Replace:
```html

>  <o-list #listAccount service="@branches@" entity="@account@" keys="@ACCOUNTID@"

Reviewer: ```suggestion
 <o-list #@attr@ service="@branches@" entity="@account@" keys="@ACCOUNTID@"
```

    columns="@ACCOUNTID;ENTITYID;OFFICEID;CDID;ANID;BALANCE;STARTDATE;ENDDATE;INTERESRATE;ACCOUNTTYP;ACCOUNTTYPEID;ACCOUNTTYPENAME@"
    controls="no" fxFlex.gt-lg="list-accounts-60" pageable="yes" fxFill>
```

<br/>

By:
```html
 <o-list #listAccount service="yourservice" entity="yourentity" keys="yourkey"
    columns="yourcolumn1;yourcolumn2"
    controls="no" fxFlex.gt-lg="list-accounts-60" pageable="yes" fxFill>
```

<br/>

> Also you need to configure your `createFilter` method located in the `grid-home.component.ts` file according the filter you want to build. Link to the filter builder documentation at the end of the readme.

Reviewer: ```suggestion
Also you need to configure your `createFilter` method located in the `list-home.component.ts` file according the filter you want to build. Link to the filter builder documentation at the end of the readme.
```

Reviewer: Creo que quedaria mejor el link a continuación para seguir el flujo de lectura porque si vas al final del readme luego tienes que volver a buscar el punto el que estabas

<br/>

> 3. Configure your filter component setting your inputs and custom ATTR to build the filter. Link to the filter builder documentation at the end of the readme.

Reviewer: ```suggestion
3. Configure your `o-filter-builder` setting your inputs and custom ATTR to build the filter. Link to the filter builder documentation at the end of the readme.
```

Reviewer: Creo que quedaria mejor el link a continuación para seguir el flujo de lectura porque si vas al final del readme luego tienes que volver a buscar el punto el que estabas

## How to use
<ol>

> <li>Download and put `list` folder in src/app/ and `i18n` into assets/  </li>

Reviewer: El markdown en el li no se renderiza, ya lo comente en mas ocasiones

<pre>
─src
    ├───app
    │   ├───list
    │   ├───...
    │   │
    ├───assets
    │   ├───css
    │   └───i18n
    |       └───en.json
    |       └───es.json
    │
    └───..

</pre>

> <li>Check if the name of the ATTR, entities and services are concording with your application</li>

Reviewer: Como se ven los <li> sin espaciados en el template manager? Revisarlo porque asi igual queda muy pegado

<li>Add the translations you want to use on your app ​​to the en.json and es.json files of your project</li>

> <li>Configure routing in `main-routing.module.ts`</li>

Reviewer: lo mismo, los markdown no se renderiza

<li>Add the module declaration in the `main.module.ts` file</li>

> <li>To configure the theme on the login component it is necessary to include `list-home.theme.scss` in `app.scss`</li>

Reviewer: NO es UN COMPONENTE DE LOGIN

<br/>

> Don't forget to customize your `o-list-item` as you want. In this template we are using a `mat-card` component but it isn't necesary.

Reviewer: ```suggestion
Don't forget to customize your `o-list-item` as you want. In this template we are using a `mat-card` component but it isn't necessary.
```
